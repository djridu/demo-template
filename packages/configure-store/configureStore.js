/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

export default function configureStore(initialState = {}, reducers = {}, middlewares = []) {
    const middlewaresToApply = [sagaMiddleware, ...middlewares];
    const enhancers = [applyMiddleware(...middlewaresToApply)];

    const composeEnhancers =
        '__DEV__' && typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
            ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
            : compose;

    const store = createStore(
        combineReducers({ ...reducers }),
        initialState,
        composeEnhancers(...enhancers),
    );

    store.runSaga = sagaMiddleware.run;
    store.injectedReducers = reducers;
    store.injectedSagas = {};

    if (module.hot) {
        module.hot.accept(() => {
            store.replaceReducer(combineReducers({ ...store.injectedReducers }));
        });
    }

    return store;
}
