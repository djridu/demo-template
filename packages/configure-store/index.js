// @flow

export { default as configureStore } from './configureStore';
