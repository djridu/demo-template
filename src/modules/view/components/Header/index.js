import React from 'react';
import { FormattedMessage } from 'react-intl';

import NavBar from './NavBar';
import HeaderLink from './HeaderLink';
import messages from './messages';

class Header extends React.Component {
    render() {
        return (
            <div>
                <NavBar>
                    <HeaderLink to="/home">
                        <FormattedMessage {...messages.home} />
                    </HeaderLink>
                    <HeaderLink to="/features">
                        <FormattedMessage {...messages.features} />
                    </HeaderLink>
                    <HeaderLink to="/">
                        <FormattedMessage {...messages.logout} />
                    </HeaderLink>
                </NavBar>
            </div>
        );
    }
}

export default Header;
