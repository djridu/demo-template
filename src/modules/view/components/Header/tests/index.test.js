import React from 'react';
import { cleanup, render } from 'react-testing-library';
import { Provider, ReactReduxContext } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter, connectRouter } from 'connected-react-router';

import { configureStore } from 'configure-store';
import history from 'utils/history';
import languageProviderReducer from 'containers/LanguageProvider/reducer';
import Header from '../index';

describe('<Header />', () => {
    const reducers = {
        language: languageProviderReducer,
        router: connectRouter(history),
    };
    const store = configureStore({}, reducers, []);

    afterEach(cleanup);

    it('should render a div', () => {
        const { container } = render(
            <Provider store={store} context={ReactReduxContext}>
                <IntlProvider locale="en">
                    <ConnectedRouter history={history}>
                        <Header />
                    </ConnectedRouter>
                </IntlProvider>
            </Provider>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
