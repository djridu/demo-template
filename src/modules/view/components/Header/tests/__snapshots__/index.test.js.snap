// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a div 1`] = `
<div>
  <div
    class="sc-1h977xf-0 cCHFNU"
  >
    <a
      class="sc-6x2adn-0 iQCTwQ"
      href="/home"
    >
      <span>
        Home
      </span>
    </a>
    <a
      class="sc-6x2adn-0 iQCTwQ"
      href="/features"
    >
      <span>
        Features
      </span>
    </a>
    <a
      class="sc-6x2adn-0 iQCTwQ"
      href="/"
    >
      <span>
        Logout
      </span>
    </a>
  </div>
</div>
`;
